RecoLocalCalo/HcalRecProducers/src/ZdcHitReconstructor.cc

```diff
@@ -97,8 +98,14 @@ void ZdcHitReconstructor::produce(edm::Event& e, const edm::EventSetup& eventSet
   
    if (det_==DetId::Calo && subdet_==HcalZDCDetId::SubdetectorId) {
      edm::Handle<ZDCDigiCollection> digi;
-     e.getByToken(tok_input_,digi);
+     e.getByToken(tok_input_hcal,digi);
      
+     if(digi->size() == 0) {
+       e.getByToken(tok_input_castor,digi);
+       if(digi->size() != 0) std::cout<<"No ZDC info in hcalDigis, using castorDigis. This is normal for all Run2 datasets"<<std::endl;
```

Comment:
cout is not allowed in production code.

I think this check should be done only when we really expect ZDC, which is probably just the HeavyIons scenario,
but also likely the pp setup for pPb processing.
For the former, in  Configuration/StandardSequences/python/ReconstructionHeavyIons_cff.py, and for the latter in RecoHI/Configuration/python/customise_PPwithHI.py.
You can also just change this to edm::LogInfo uniformly and then hope that someone will look at LogInfo (this severity level is not displayed/logged in production).